### PR TITLE
Refactor GeodeTypeIds enums...

### PR DESCRIPTION
**Pull request by austxcodemonkey**
… into DSCode, DSFid, and InternalId enums, make enums enum classes

Co-authored-by: Ivan Godwin <igodwin@pivotal.io>
Co-authored-by: Jacob Barrett <jbarrett@pivotal.io>

### CODE REVIEW

File: cppcache/include/geode/internal/DSCode.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+//
+// Created by Pivotal on 8/6/18.
```
Comment (echobravopapa) [inline cppcache/include/geode/internal/DSCode.hpp]: Need the Apache header

File: cppcache/include/geode/internal/DSFixedId.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+//
+// Created by Pivotal on 8/6/18.
```
Comment (echobravopapa) [inline cppcache/include/geode/internal/DSFixedId.hpp]: header

File: cppcache/include/geode/internal/InternalId.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
+//
+// Created by Pivotal on 8/6/18.
```
Comment (echobravopapa) [inline cppcache/include/geode/internal/InternalId.hpp]: again

**Comment by austxcodemonkey** [comment]
Hit CLI breaks in Windows after applying a small requested change from JB, so closing this while I fix things.

**Pull request merged**